server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts

```diff
@@ -136,149 +133,166 @@ const saveClinicalDetails = (request: Request, response: Response, isolationAddr
-            saveClinicalDetailsFieldsLogger.info('got response from the DB for address creation', Severity.LOW);
-            saveClinicalDetails(request, response, result.data.insertAndGetAddressId.integer);
-        }).catch(err => {
-            saveClinicalDetailsFieldsLogger.error(`got errors approaching the graphql API ${err}`,Severity.HIGH);
-            response.status(errorStatusCode).send(err);
+    const isolationAddress = request.body.clinicalDetails?.isolationAddress;
+    if (isolationAddress) {
+        const requestAddress: InsertAndGetAddressIdInput = formatToInsertAndGetAddressIdInput(isolationAddress);
+        const saveAddressLogger = logger.setup({...logData, workflow: `${logData.workflow}: saving isolation address`})
+
+        const parameters = {input: JSON.stringify(requestAddress)};
+        saveAddressLogger.info(launchingDBRequestLog(parameters), Severity.LOW);
+
+        graphqlRequest(CREATE_ADDRESS, response.locals, {input: requestAddress})
+        .then((result: CreateAddressResponse) => {
+            saveAddressLogger.info(validDBResponseLog, Severity.LOW);
+            saveClinicalDetails(request, response, logData, result.data.insertAndGetAddressId.integer);
+        }).catch(error => {
+            saveAddressLogger.error(invalidDBResponseLog(error),Severity.HIGH);
+            response.status(errorStatusCode).send(error);
         });
     } else {
-        saveClinicalDetails(request, response, null);
+        saveClinicalDetails(request, response, logData);
     }
 });
 
 clinicalDetailsRoute.get('/coronaTestDate', (request: Request, response: Response) => {
     const coronaTestDateLogger = logger.setup({
-        workflow: 'Getting corona test date of patient',
+        workflow: 'query test date and symptoms data of patient',
         investigation: response.locals.epidemiologynumber,
         user: response.locals.user.id
     });
-    coronaTestDateLogger.info(`launcing DB request with parameter ${request.params.investigationId}`, Severity.LOW);
-    graphqlRequest(GET_CORONA_TEST_DATE_OF_PATIENT, response.locals, { currInvestigation: Number(response.locals.epidemiologynumber) })
-        .then((result: CoronaTestDateQueryResult) => {
-            coronaTestDateLogger.info('got response from DB', Severity.LOW);
-            response.send(result.data.allInvestigations.nodes[0]);
-        }).catch(err => {
-            coronaTestDateLogger.error(`got errors approaching the graphql API ${err}`, Severity.HIGH);
-            response.status(errorStatusCode).send(err);
-        });
+    const parameters = {currInvestigation: +response.locals.epidemiologynumber};
+    coronaTestDateLogger.info(launchingDBRequestLog(parameters), Severity.LOW);
+    graphqlRequest(GET_CORONA_TEST_DATE_OF_PATIENT, response.locals, parameters)
+    .then((result: CoronaTestDateQueryResult) => {
+        coronaTestDateLogger.info(validDBResponseLog, Severity.LOW);
+        response.send(result.data.allInvestigations.nodes[0]);
```

Comment:
I know It's not your query but maybe its better to change GET_CORONA_TEST_DATE_OF_PATIENT to something like `investigationByEpidemiologyNumber(epidemiologyNumber: $)` instead of `allInvestigations(condition: {epidemiologyNumber: $})`? it only needs to return 1 item anyway 🤷‍♂️

Reply:
I wanted to delete this route in a separate pr, so I won’t change it for now